tests/test_examples.py

```diff
@@ -8,57 +8,62 @@
+    ('vae/ss_vae_M2.py', ['--num-epochs=1', '--use-cuda']),
+    ('vae/ss_vae_M2.py', ['--num-epochs=1', '--aux-loss', '--use-cuda']),
+    ('vae/ss_vae_M2.py', ['--num-epochs=1', '--enum-discrete=parallel', '--use-cuda']),
+    ('vae/ss_vae_M2.py', ['--num-epochs=1', '--enum-discrete=sequential', '--use-cuda']),
+]
 
-def discover_examples():
+CPU_EXAMPLES.sort()
+CUDA_EXAMPLES.sort()
+
+
+def make_ids(examples):
+    return ['{} {}'.format(example, ' '.join(args)) for example, args in examples]
+
+
+def test_coverage():
+    cpu_tests = set([name for name, _ in CPU_EXAMPLES])
+    cuda_tests = set([name for name, _ in CUDA_EXAMPLES])
     for root, dirs, files in os.walk(EXAMPLES_DIR):
         for basename in files:
             if not basename.endswith('.py'):
                 continue
             path = os.path.join(root, basename)
             with open(path) as f:
                 text = f.read()
-            if '--num-epochs' in text:
-                args = ['--num-epochs=1']
-            elif '--num-steps' in text:
-                args = ['--num-steps=1']
-            elif '--num-samples' in text:
-                args = ['--num-samples=1']
-            else:
-                # Either this is not a main file, or we don't know how to run it cheaply.
-                continue
             example = os.path.relpath(path, EXAMPLES_DIR)
-            # TODO: May be worth whitelisting the set of arguments to test
-            # for each example.
-            CPU_EXAMPLES.append((example, args))
-            cuda_args = args + ['--cuda']
-            if '--cuda' in text:
-                CUDA_EXAMPLES.append((example, cuda_args))
-            if '--aux-loss' in text:
-                CPU_EXAMPLES.append((example, args + ['--aux-loss']))
-                if '--cuda' in text:
-                    CUDA_EXAMPLES.append((example, cuda_args + ['--aux-loss']))
-            if '--enum-discrete' in text:
-                CPU_EXAMPLES.append((example, args + ['--enum-discrete=sequential']))
-                CPU_EXAMPLES.append((example, args + ['--enum-discrete=parallel']))
-                if '--cuda' in text:
-                    CUDA_EXAMPLES.append((example, cuda_args + ['--enum-discrete=sequential']))
-                    CUDA_EXAMPLES.append((example, cuda_args + ['--enum-discrete=parallel']))
-            if '--num-iafs' in text:
-                CPU_EXAMPLES.append((example, args + ['--num-iafs=1']))
-                if '--cuda' in text:
-                    CUDA_EXAMPLES.append((example, cuda_args + ['--num-iafs=1']))
-    CPU_EXAMPLES.sort()
-    CUDA_EXAMPLES.sort()
-
-
-discover_examples()
-
-
-def make_ids(examples):
-    return ['{} {}'.format(example, ' '.join(args)) for example, args in examples]
+            if '__main__' in text:
+                if example not in cpu_tests:
+                    pytest.fail('Example: {} not covered in CPU_TESTS.'.format(example))
+                if 'cuda' in text and example not in cuda_tests:
```

Comment:
nit: it's safer to check for `'--cuda' in text` because there might be mention of cuda in the comment section of non-cuda examples, but `--cuda` is universally used to turn on cuda support in the examples

Reply:
yeah fishery models are pretty common in the hierarchical bayes literature, and a barracuda might sneak in to a cpu-only example.

Reply:
Haha..actually we have examples that use `--use-cuda` or even `--no-cuda` which we were missing out on. Hence this check in coverage. We can standardize these examples, but there is no guarantee someone might not end up sneaking in a new variant like `--go-cuda`.

Reply:
>actually we have examples that use --use-cuda

oof thats terrible we should standardize them

Reply:
Yeah, we should. This coverage is designed to catch all potential examples that are cuda compatible, but I suppose we should insist on uniformity of arguments during code reviews.